ibis/pandas/udf.py

```diff
@@ -165,30 +173,49 @@ def udf_signature(input_type, klass):
     >>> import pandas as pd
     >>> from pandas.core.groupby import SeriesGroupBy
     >>> import ibis.expr.datatypes as dt
-    >>> input_type = [dt.int64, dt.double]
-    >>> sig = udf_signature(input_type, pd.Series)
+    >>> input_type = [dt.string, dt.double]
+    >>> sig = udf_signature(input_type, pin=None, klass=pd.Series)
     >>> pprint(sig)  # doctest: +ELLIPSIS
-    ((<class '...Series'>,
-      <... 'int'>,
-      <... 'numpy.integer'>,
-      <... 'NoneType'>),
+    ((<class '...Series'>, <... '...str...'>, <... 'NoneType'>),
      (<class '...Series'>,
       <... 'float'>,
       <... 'numpy.floating'>,
       <... 'NoneType'>))
-    >>> input_type = [dt.Int64(nullable=False), dt.Double(nullable=False)]
-    >>> sig = udf_signature(input_type, SeriesGroupBy)
+    >>> not_nullable_types = [
+    ...     dt.String(nullable=False), dt.Double(nullable=False)]
+    >>> sig = udf_signature(not_nullable_types, pin=None, klass=pd.Series)
     >>> pprint(sig)  # doctest: +ELLIPSIS
-    ((<class '...SeriesGroupBy'>,
-      <... 'int'>,
-      <... 'numpy.integer'>),
-     (<class '...SeriesGroupBy'>,
+    ((<class '...Series'>, <... '...str...'>),
+     (<class '...Series'>,
       <... 'float'>,
       <... 'numpy.floating'>))
+    >>> sig0 = udf_signature(input_type, pin=0, klass=SeriesGroupBy)
+    >>> sig1 = udf_signature(input_type, pin=1, klass=SeriesGroupBy)
+    >>> pprint(sig0)  # doctest: +ELLIPSIS
+    (<class '...SeriesGroupBy'>,
+     (<class '...SeriesGroupBy'>,
+      <... 'float'>,
+      <... 'numpy.floating'>,
+      <... 'NoneType'>))
+    >>> pprint(sig1)  # doctest: +ELLIPSIS
+    ((<class '...SeriesGroupBy'>,
+      <... '...str...'>,
```

Comment:
The surrounding ellipses are here because in Python 2 this is `basestring` (which is the base class of both `str` and `unicode` in Python 2).